basic_module/staking/src/impls.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::check::check;
+use crate::core::{Abci, AdditionalTxCreator, StakingView};
+use crate::error::Error;
+use crate::execute::{apply_internal, execute_auto_action};
+use crate::state::{get_stakes, Banned, CurrentValidators, Metadata, Params};
+use crate::transactions::{
+    create_close_block_transactions, create_open_block_transactions, SignedTransaction, Transaction,
+};
+use crate::types::{Tiebreaker, Validator};
+use coordinator::types::{ExecuteTransactionError, HeaderError, TransactionExecutionOutcome, VerifiedCrime};
+use coordinator::Header;
+use fkey::Ed25519Public as Public;
+use std::cell::RefCell;
+use std::collections::HashMap;
+
+struct ABCIHandle {
+    executing_block_header: RefCell<Header>,
+}
+
+impl AdditionalTxCreator for ABCIHandle {
+    fn create(&self) -> Vec<Transaction> {
+        let mut transactions = create_open_block_transactions();
```

Comment:
The original code in the mold branch is as below.
```rust
create_open_block_transactions()
    .into_iter()
    .chain(create_close_block_transactions(&*self.executing_block_header.borrow()).into_iter())
    .collect()
```
I changed it because the code above is inefficient.